src/Repo_new.java

```diff
@@ -0,0 +1,804 @@
+           "{0}\n\n"
+         + FOOTER_START + '\n'
+         + "{1}\n"
+         + FOOTER_END);
+      //@formatter:on
+    }
+
+    public Server getIdleConnection() {
+        return _pool.get();
+    }
+
+    public void releaseConnection(final Server s) {
+        _pool.release(s);
+    }
+
+    private Server createNewConnection() {
+        final StarTeamURL url = getUrl();
+        final RepoProperties props = getRepoProperties();
+        final int port = Integer.parseInt(url.getPort());
+        final int protocol = url.getProtocol();
+        final String host = url.getHostName();
+        final String ca = props.getProperty(PROP_CA, null);
+        final ServerInfo info = new ServerInfo();
+        String userName = props.getProperty(PROP_USER, null);
+        String password = props.getProperty(PROP_PASSWORD, null);
+        final Server server;
+
+        if (userName == null) {
+            userName = url.getUserName();
+        }
+        if (password == null) {
+            password = url.getPassword();
+        }
+        if (ca != null) {
+            if (Boolean.parseBoolean(ca)) {
+                info.setAutoLocateCacheAgent(true);
+            } else {
+                final int ind = ca.indexOf(':');
+
+                if (ind == -1) {
+                    info.setMPXCacheAgentAddress(ca);
+                } else {
+                    info.setMPXCacheAgentAddress(ca.substring(0, ind));
+                    info.setMPXCacheAgentPort(Integer.parseInt(ca
+                            .substring(ind + 1)));
+                }
+            }
+        }
+
+        info.setHost(host);
+        info.setPort(port);
+        info.setCompression(true);
+        info.setConnectionType(protocol);
+        info.setMPXCacheAgentThreadCount(Integer.parseInt(props.getProperty(
+                PROP_CATHREADS, PROP_DEFAULT_CATHREADS)));
+        info.setEnableCacheAgentForFileContent(ca != null);
+        info.setEnableCacheAgentForObjectProperties(ca != null);
+        server = new Server(info);
+        server.connect();
+
+        if ((userName != null) && (password != null)) {
+            if (server.logOn(userName, password) == 0) {
+                throw new ConfigurationException("Failed to login to " + host
+                        + ':' + port + " as user " + userName);
+            }
+            cacheLogOnCredentials(server, userName, password);
+        } else if (autoLogOn(server) == 0) {
+            final CredentialHelper h = getGit().getCredentialHelper(
+                    getProtocol(url), host, userName);
+
+            if (h != null) {
+                final boolean[] approved = new boolean[1];
+                h.getCredentials(new CredentialCallBack() {
+                    @Override
+                    public boolean approve(final String user,
+                            final String password) {
+                        try {
+                            if (server.logOn(user, password) != 0) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS134)